Update options.md (#3021)
gohugoio/hugoDocs: content/en/_common/functions/js/options.md

  ```

target

> : (`string`) The language target. One of: `es5`, `es2015`, `es2016`, `es2017`, `es2018`, `es2019`, `es2020`, `es2021`, `es2022`, `es2023`, `es2024`, or `esnext`. The default is `esnext`.

Reviewer: ```suggestion
: (`string`) The language target. One of: `es5`, `es2015`, `es2016`, `es2017`, `es2018`, `es2019`, `es2020`, `es2021`, `es2022`, `es2023`, `es2024`, or `esnext`. Default is `esnext`.
```

Reviewer: For right or wrong, "Default is something" is standard throughout the documentation. If we change this at some point, we need to change it everywhere. But please do not submit a PR for that now... there's a very large change set in progress and I don't want to deal with a hundred merge conflicts.

platform
: {{< new-in 0.140.0 />}}

> : (`string`) One of `browser`, `node`, `neutral`. The default is `browser`. See https://esbuild.github.io/api/#platform.

Reviewer: ```suggestion
: (`string`) One of `browser`, `node`, `neutral`. Default is `browser`. See https://esbuild.github.io/api/#platform.
```

Reviewer: See previous comment.

externals
: (`slice`) External dependencies. Use this to trim dependencies you know will never be executed. See https://esbuild.github.io/api/#external.

defines

> : (`map`) This option allows to define a set of string replacements to be performed when building. It should be a map where each key will be replaced by its value.

Reviewer: ```suggestion
: (`map`) This option allows you to define a set of string replacements to be performed when building. It must be a map where each key will be replaced by its value.
```

: See https://esbuild.github.io/api/#drop

sourceMap
: (`string`) Whether to generate `inline`, `linked`, or `external` source maps from esbuild. Linked and external source maps will be written to the target with the output file name + ".map". When `linked` a `sourceMappingURL` will also be written to the output file. By default, source maps are not created. Note that the `linked` option was added in Hugo 0.140.0.

sourcesContent
: {{< new-in 0.140.0 />}}
: (`bool`) Whether to include the content of the source files in the source map. By default, this is `true`.

JSX
: {{< new-in 0.124.0 />}}

> : (`string`) How to handle/transform JSX syntax. One of: `transform`, `preserve`, `automatic`. The default is `transform`. Notably, the `automatic` transform was introduced in React 17+ and will cause the necessary JSX helper functions to be imported automatically. See https://esbuild.github.io/api/#jsx.

Reviewer: ```suggestion
: (`string`) How to handle/transform JSX syntax. One of: `transform`, `preserve`, `automatic`. Default is `transform`. Notably, the `automatic` transform was introduced in React 17+ and will cause the necessary JSX helper functions to be imported automatically. See https://esbuild.github.io/api/#jsx.
```
